api/provision/start-provision-job.ts

```diff
@@ -0,0 +1,101 @@
+import { sfnClient } from "../../utils/aws-sdk/step-functions";
+import { APIGatewayProxyEvent, APIGatewayProxyResult, Context } from "aws-lambda";
+import { ApiSuccessResponse, ErrorStatusCode, OtherErrorResponse, SuccessStatusCode } from "../../utils/response";
+import {
+  CspInvocation,
+  ProvisionRequest,
+  provisionRequestSchema,
+  ProvisionRequestType,
+} from "../../models/provisioning-jobs";
+import { wrapSchema } from "../../utils/middleware/schema-wrapper";
+import { errorHandlingMiddleware } from "../../utils/middleware/error-handling-middleware";
+import middy from "@middy/core";
+import validator from "@middy/validator";
+import jsonBodyParser from "@middy/http-json-body-parser";
+import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
+import cors from "@middy/http-cors";
+import xssSanitizer from "../../utils/middleware/xss-sanitizer";
+import { IpCheckerMiddleware } from "../../utils/middleware/ip-logging";
+import { CloudServiceProvider } from "../../models/cloud-service-providers";
+import { HttpMethod } from "../../lib/http";
+import { CORS_CONFIGURATION, MIDDY_CORS_CONFIGURATION } from "../../utils/cors-config";
+import { REQUEST_BODY_INVALID } from "../../utils/errors";
+import { cspPortfolioIdChecker } from "../../utils/middleware/check-csp-portfolio-id";
+
+const SFN_ARN = process.env.SFN_ARN ?? "";
+
+/**
+ * Starts provisioning request from SNOW and starts the Step Function execution
+ *
+ * @param event - POST request from API Gateway with provisioning job properties
+ */
+export async function baseHandler(event: APIGatewayProxyEvent, context: Context): Promise<APIGatewayProxyResult> {
+  try {
+    const cspInvocationJob = transformProvisionJob(event.body as any);
+    console.log("SentToSfn: " + JSON.stringify(cspInvocationJob));
+
+    // starting the execution
+    const result = await sfnClient.startExecution({
+      input: JSON.stringify(cspInvocationJob),
+      stateMachineArn: SFN_ARN,
+    });
```

Comment:
It did cross my mind, but are you able to add specific logic in API GW to transform the request? I know there is mapping capabilities, but I wasn't sure the extend of them. 

Since we were already using the lambda to transform the incoming request previously, I decide to keep using the lambda to invoke the step function rather than directly from API GW. I'm open to look into it more though as that would reduce the need to use a lambda.

Reply:
Almost definitely but unless there are any significant issues I'd ask that we punt this to a separate ticket so that we can get the rest of the functions pushed through.

Reply:
Made note about future ticket

Reply:
Ticket for the above is `AT-7177`